Reorganize getting-started + added a new tab "Core Concepts" (#24)
php-etl/documentation: content/getting-started/installation/_index.en.md

---
title: "Installation"
date: 2021-10-12T15:21:02+02:00
draft: false
type: "getting-started"
shortDescription: Getting started
weight: 2
---

> The first thing you have to do is to create a folder in which you will put your project.
> In a terminal, type the following command:

Reviewer: ```suggestion
Starting from here we will work inside a project directory we will name `my_project_directory`.
```

```shell
mkdir my_project_directory && cd my_project_directory
```

> After creating your project, you will have to initialize your project and add all the dependencies you need.

Reviewer: ```suggestion
Once the project dir is created and you have changed your current directory to it, the first task will be to initialize a composer project inside this directory and add all required dependencies.
```

```shell
composer init
```

> > Notice: You must choose `dev` as the value for the `minimum-stability` option.

Reviewer: **🎯 suggestion:** proposes une commande `composer init` plus complète avec les options appropriées pour que l'utilisateur n'ait pas à chercher ce genre d'avertissement.‏

> When asked if you want to define your require and require-dev dependencies interactively, answer `no`.

> In the auto-generated `composer.json` file, add the following lines at the end :

Reviewer: **💭 thought:** il faudrait que l'on intègre ces changements dans la commande `satellite`‏

```json
{
  // ...
  "config": {
    "bin-dir": "bin"
  },
  "scripts": {
    "post-install-cmd": [
      "Kiboko\\Component\\Satellite\\ComposerScripts::postInstall"
    ],
    "post-update-cmd": [
      "Kiboko\\Component\\Satellite\\ComposerScripts::postUpdate"
    ]
  }
}
```

> The most important thing you need to add to your project is our compiler :

Reviewer: ```suggestion
Now, your environment is ready for the satellite compiler installation. The following command will install it.
```
